lib/travis/build/script/hack.rb

```diff
@@ -0,0 +1,68 @@
+module Travis
+  module Build
+    class Script
+      class Hack < Php
+        DEFAULTS = {
+          hhvm: 'hhvm', # lts
+          php: '7.2',
+        }
+
+        VALID_HHVM = %w(
+          hhvm
+          hhvm-dbb
```

Comment:
- I'm guessing this a typo of debug
- HHVM now follows standard-ish debian packaging: -dbg packages are just debug symbols for the main package - they're no longer include debug executables. It doesn't really make sense to select them /instead of/ a non-debug package.
- it was possible to specify arbitrary versions, e.g. '3.21-lts' - is it possible to bring this back rather than use the whitelist?

Reply:
These values are taken from the official installation guide at https://docs.hhvm.com/hhvm/installation/linux#other-packages. (Though, as you correctly point out, `dbb` is a typo of `dbg`.)

Reply:
I'll look into LTS. However, I'll point out that it is rather unwieldy to validate the user input; https://docs.hhvm.com/hhvm/installation/release-schedule#Lifecycle says 3.27 and 3.30 are LTS, but 3.27 is not, and we have no idea how 3.28 behaves. So we will have to rely on users to supply sensible values.

Reply:
3.27 is an LTS and is supported
3.28 and 3.29 are unsupported, I’ll update that